### PR TITLE
chore/dedicated-trigger-of-downstream-jobs

### PR DESCRIPTION
**Pull request by jimid27**
- Relates: https://github.com/PrefectHQ/platform/issues/6336
- Needs: https://github.com/PrefectHQ/actions-release-ui-components/pull/12
- Needs: https://github.com/PrefectHQ/actions-trigger-downstream-npm-package-updates/pull/1
- Needs: https://github.com/PrefectHQ/vue-compositions/pull/411, https://github.com/PrefectHQ/vue-charts/pull/341, https://github.com/PrefectHQ/prefect-design/pull/1174, https://github.com/PrefectHQ/graphs/pull/479
- Triggers all relevant downstream npm update jobs upon release of this package

### CODE REVIEW

**Comment by collincchoy** [comment]
hey FYI @jimid27 , this repo/package is a bit weird compared to the others. For example, there is no build script in here so `npm run build` fails. it's basically just a config file that's version controlled. 

Interestingly in terms of impact, this is probably at the tippy top of the our dependency chain tho so it actually is the most friction to bump _everywhere_ and it isn't bumped often.

**Comment by collincchoy** [comment]
I think the workflow used to just be a straight publish. There are definitely other upstream npm package dependencies here too so i'd guess like, setup node -> npm ci -> publish is fine. 

i see that you're breaking out the upstream version bumps from the shared action which I imagine should make having a little custom gha workflow here easier? so we could just fallback to a previous version of the `release` workflow that's just a "basic checkout -> setup node -> install deps -> publish package"  -> fan out trigger downstream version bumping jobs? 

we could instead try to tweak the shared action to effectively skip the build step - i don't feel strongly on which direction to take but I'd imagine it just adds a weird, maybe-unnecessary layer of complexity into that action 🤷 .

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -18,8 +18,24 @@ jobs:
...
+          DOWNSTREAM_REPO_NAME: nebula-ui
+          RELEASE_TAG: ${{ steps.release-ui-components.publish.outputs.version}}
+
+      - uses: prefecthq/actions-trigger-downstream-npm-package-updates@main
+        id: trigger-downstream-npm-package-update-prefect-oss
+        if: ${{ steps.release-ui-components.publish.outputs.type }}
+        with:
+          GITHUB_TOKEN: ${{ secrets.PREFECT_ACTIONS_RW }}
+          DOWNSTREAM_REPO_NAME: prefect
+          RELEASE_TAG: ${{ steps.release-ui-components.publish.outputs.version}}
```
Comment (collincchoy) [inline .github/workflows/release.yml:41]: Is this action at a point that we could plop in _all_ of the other downstream packages (pretty much all of our npm packages use this) so
- prefect-design
- prefect-ui-library
- graphs
- vue-compositions

or do we also need separate secrets for each still?

(😅 - i'd like to cut a new version of this "soon". suuuuuper no rush tho. "soon" could be really whenever, but it'd be neat to test all of this out with the next version bump)

Comment (jimid27) [reply .github/workflows/release.yml:41]: We would need to create GH secrets still but I can get that done today/monday!

Comment (collincchoy) [reply .github/workflows/release.yml:41]: that would be dope! 

will save me _so_ many button clicks!!!

**Review by collincchoy — APPROVED** [review]

**Pull request merged**